metaflow/includefile.py

```diff
@@ -8,259 +9,213 @@
+            try:
+                value = json.loads(value)
+            except json.JSONDecodeError as e:
+                raise MetaflowException(
+                    "IncludeFile '%s' (value: %s) is malformed" % (param.name, value)
+                )
+            # All processing has already been done so we just convert to a
+            # IncludedFile
+            return IncludedFile(value)
+
+        path = os.path.expanduser(value)
+
+        prefix_pos = path.find("://")
+        if prefix_pos > 0:
+            # Scenario B.1
+            raise MetaflowException(
+                "IncludeFile using a direct reference to a file in cloud storage is no "
+                "longer supported. Contact the Metaflow team if you need this supported"
+            )
+            # if DATACLIENTS.get(path[:prefix_pos]) is None:
+            #     self.fail(
+            #         "IncludeFile: no handler for external file of type '%s' "
+            #         "(given path is '%s')" % (path[:prefix_pos], path)
+            #     )
+            # # We don't need to do anything more -- the file is already uploaded so we
+            # # just return a blob indicating how to get the file.
+            # return IncludedFile(
+            #     CURRENT_UPLOADER.encode_url(
+            #         "external", path, is_text=self._is_text, encoding=self._encoding
+            #     )
+            # )
         else:
-            # We will just store the URL in the datastore along with text/encoding info
-            return lambda is_text=self._is_text, encoding=self._encoding, value=value: Uploader.encode_url(
-                "external", value, is_text=is_text, encoding=encoding
+            # Scenario B.2
+            # Check if this is a valid local file
+            try:
+                with open(path, mode="r") as _:
+                    pass
+            except OSError:
+                self.fail("IncludeFile: could not open file '%s' for reading" % path)
+            handler = DATACLIENTS.get(ctx.ds_type)
+            if handler is None:
+                self.fail(
+                    "IncludeFile: no data-client for datastore of type '%s'"
+                    % ctx.ds_type
+                )
+
+            # Now that we have done preliminary checks, we will delay uploading it
+            # until later (so it happens after PyLint checks the flow but we prepare
+            # everything for it)
+            lambda_ctx = _DelayedExecContext(
+                flow_name=ctx.flow_name,
+                path=path,
+                is_text=self._is_text,
+                encoding=self._encoding,
+                handler_type=ctx.ds_type,
+                echo=ctx.logger,
+            )
+
+            def _delayed_eval_func(ctx=lambda_ctx, return_json=False):
+                incl_file = IncludedFile(
+                    CURRENT_UPLOADER.store(
+                        ctx.flow_name,
```

Comment:
After fixing bug on line 217, (Invalid return of `DelayedEvaluationParameter`), `ctx` is `False`